examples/templates/docker-with-dotfiles/main.tf

```diff
@@ -0,0 +1,118 @@
+terraform {
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = "0.4.1"
+    }
+    docker = {
+      source  = "kreuzwerker/docker"
+      version = "~> 2.16.0"
+    }
+  }
+}
+
+# Admin parameters
+
+# Comment this out if you are specifying a different docker
+# host on the "docker" provider below.
+variable "step1_docker_host_warning" {
+  description = <<-EOF
+  This template will use the Docker socket present on
+  the Coder host, which is not necessarily your local machine.
+
+  You can specify a different host in the template file and
+  surpress this warning.
+  EOF
+  validation {
+    condition     = contains(["Continue using /var/run/docker.sock on the Coder host"], var.step1_docker_host_warning)
+    error_message = "Cancelling template create."
+  }
+
+  sensitive = true
+}
```

Comment:
The Docker provider will use `DOCKER_HOST` by default, so we should be able to just exclude it from the provider and have this _just work_.

Reply:
Sadly, many systems do not have `DOCKER_HOST` set

Reply:
However, I removed this warning since this just demonstrates dotfiles usage